modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/localizedObjectFields/SelectLocalizedObjectField.tsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayInput} from '@clayui/form';
+import {stringUtils} from '@liferay/object-js-components-web';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import FieldBase from '../FieldBase/ReactFieldBase.es';
+import SingleSelectBase from '../Select/SingleSelectBase';
+import {useNormalizedOptionsMemo} from '../Select/hooks';
+import {SelectMainProps} from '../Select/select.d';
+import {LocalizedValue} from '../types';
+import {isEmptyObject} from '../util/basicJsUtils';
+import LocalesDropdown, {
+	AvailableLocale,
+	EditingLocale,
+} from '../util/localizable/LocalesDropdown';
+import {getEditingLocales, getLocale} from './util/locales';
+
+import './SelectLocalizedObjectField.scss';
+
+import type {Locale} from '../types';
+
+type valueTypes = {} | LocalizedValue<string>;
+
+export interface SelectLocalizedObjectFieldProps
+	extends Omit<SelectMainProps, 'value'> {
+	availableLocales: AvailableLocale[];
+	defaultLocale: AvailableLocale;
+	value: valueTypes;
+}
+
+function normalizeValues(
+	defaultLocaleId: Locale,
+	predefinedValue: string | string[] | undefined,
+	values: LocalizedValue<string[] | string>
+): LocalizedValue<string> {
+	if (isEmptyObject(values)) {
+		if (predefinedValue) {
+			return {
+				[defaultLocaleId]:
+					Array.isArray(predefinedValue) && predefinedValue.length
+						? predefinedValue[0]
+						: predefinedValue,
```

Comment:
Hey @cgmonte, if the predefined value is an empty array, we would fall into the second option of the ternary, correct? Would this be a problem in the current logic?

I considered the case of the expression builder of the predefined value returning something other than `string | string[] | undefined`

But I don't know if this is possible, was this considered or tested?

Reply:
Yeah, I think you are right, there is a problem in this logic. Is not taking care of the empty array case. I will analyse it further, thanks.